### PR TITLE
Fix inline block toolbar keyboard navigation

### LINKED ISSUE #26114: Keyboard navigation to image caption toolbar
**Describe the bug**
Cannot navigate into the image caption toolbar with the keyboard.

**To reproduce**
1. Create image block
2. Enter some text
3. Try to change that text color with just the keyboard
4. Unable to navigate into toolbar

**Expected behavior**
Should be able to navigate into the image caption toolbar with the keyboard.

**Screenshots**
caption toolbar

**Editor version (please complete the following information):**
- WordPress version: 5.5.1
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Gutenberg plugin
- If the Gutenberg plugin is installed, which version is it? Master branch (b126b3f)

**Desktop:**
 - OS: MacOS
 - Browser: Chrome
 - Version: 85.0.4183.121

### PR DESCRIPTION
**Pull request by diegohaz**
Closes #26114

## Description

This PR makes the inline toolbars (present on Image, Audio and Video captions for example) accessible through the keyboard. Just like the block toolbars, now it's possible to press <kbd>Shift+Tab</kbd> or <kbd>Alt+F10</kbd> to access the nested toolbar.

## How has this been tested?

Check Image, Audio, Video, Embed and Gallery (including the gallery image) captions and try to access the caption toolbar with the keyboard by pressing <kbd>Shift+Tab</kbd> or using screen reader tools to find the toolbar.

I've also included an e2e test on the Image block.

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-editor/src/components/rich-text/format-toolbar-container.js [hunk 1]
```diff
@@ -19,9 +24,12 @@ const FormatToolbarContainer = ( { inline, anchorRef } ) => {
 				focusOnMount={ false }
 				anchorRef={ anchorRef }
 				className="block-editor-rich-text__inline-format-toolbar"
-				__unstableSlotName="block-toolbar"
+				// Render inline
+				__unstableSlotName={ null }
```
Comment (diegohaz) [inline packages/block-editor/src/components/rich-text/format-toolbar-container.js:28]: I'm passing `null` to the slot name so it renders inline in the React tree, and not on a separate slot as it won't find any slot: https://github.com/WordPress/gutenberg/blob/5c24a7d6e5882a1f2b9e0dff787fecfccce01335/packages/components/src/popover/index.js#L627-L635

I don't know if there's a better way to achieve this, but I think we could have an `inline` prop or something on Popover.

**Comment by alexstine** [comment]
There is only one improvement I will suggest.

Can you modify the PR to allow Alt+F10 open the formatting toolbar if focus in in Caption field? Currently, only Shift+Tab can do this.

Otherwise it is functioning well.

Thanks.

**Comment by diegohaz** [comment]
@alexstine Thanks for reviewing it! I've added support for Alt+F10.

**Comment by alexstine** [comment]
@diegohaz I've given it a test, this now seems perfect.

Thanks.

**Review by ellatrix — CHANGES_REQUESTED** [review]
Hm, I don't think we should put popovers inline in the editor canvas. Would be great to find another way to do this.

**Comment by diegohaz** [comment]
@ellatrix Thanks for reviewing this. The only other way I can think of is [creating focus trap elements](https://twitter.com/diegohaz/status/1229837604977348612) that will transfer the focus between the image caption editable content and the toolbar and vice-versa. But this technique is rather complex and there are [known issues with VoiceOver](https://twitter.com/diegohaz/status/1252633190956781570) (VO will just ignore these focus trap elements when navigating through elements with VO keys unless you have an unreliable timeout on it).

The safest way I can think of is relying on the DOM order like in this PR. What do you think is the caveat on this?

**Comment by ellatrix** [comment]
It should align more with how the block toolbar works, which is also not embedded in the content's DOM. Ideally, the UI DOM and the content DOM should be kept separate so the styling is isolated. It's also possible to run into some weird selector issues like :first-child and :last-child if the popover unexpectedly appears in the content.

To align better with the block toolbar, I'd imagine you could reverse tab into the inline toolbar just like you would for the block toolbar. Another reverse tab could put you in the block toolbar. In other to accomplish that, the inline toolbar DOM should be rendered right after the block toolbar DOM.

**Comment by ellatrix** [comment]
I created #29874. The inline toolbar is actually accessible by keyboard, it's just misplaced. It should be rendered after the block toolbar instead of before.

**Comment by ajlende** [comment]
Fixed in #31134

**Pull request closed without merging**